### DISCUSSION TITLE
ChatGPT 的資料安全性

### DISCUSSION THREAD

**Discussion by pudding0803** [1101-GS4719-C/Python-Community-GS4719-C#1455] [科技新知]
不知道大家有沒有試過用 ChatGPT 幫助生成程式碼，我個人是覺得滿方便的！雖然他常常自己產生錯誤的回答，或是有無法得知近期資訊等問題（因為他在 2021/09 就停止接收資訊了），但只要學著自己判斷，結合自己的知識、網路上的文件與論壇，ChatGPT 無疑是一個非常方便的工具。

不過我最近看到了幾個關於公司禁用 ChatGPT 的新聞：
- [三星電子宣布禁止員工使用ChatGPT類工具](https://tw.news.yahoo.com/%E4%B8%89%E6%98%9F%E9%9B%BB%E5%AD%90%E5%AE%A3%E5%B8%83%E7%A6%81%E6%AD%A2%E5%93%A1%E5%B7%A5%E4%BD%BF%E7%94%A8chatgpt%E9%A1%9E%E5%B7%A5%E5%85%B7-012753328.html)
- [防機密外洩！蘋果限制員工用ChatGPT等工具](https://tw.sports.yahoo.com/news/%E9%98%B2%E6%A9%9F%E5%AF%86%E5%A4%96%E6%B4%A9-%E8%98%8B%E6%9E%9C%E9%99%90%E5%88%B6%E5%93%A1%E5%B7%A5%E4%BD%BF%E7%94%A8chatgpt%E7%AD%89ai%E5%B7%A5%E5%85%B7-082021706.html)

關於這個議題，我先放上與當事人的對話：
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/45879162/12b674bb-7413-48e1-bf8b-ad6f6d314104)
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/45879162/21c17996-be5f-446e-8f66-0cfaf1013e82)

可以發現的是，連 ChatGPT 都不認為上傳敏感資料是安全的，甚至可以說是連他也不建議。不過我個人認為，如果在遇到大問題時可以先拆解成小問題，並只詢問 ChatGPT 一些小問題，而非直接將商業問題全盤貼上的話，或許是一個不錯的選擇。但要判斷員工是否具備這樣的能力可以說是非常主觀且困難的，因此就現階段而言，先禁止在商業上使用應該是最好的做法。

**Comment by Aiame** [comment]
Chatgpt在幫助寫程式上確實好用，也存在著錯誤訊息，之前曾經試著問他怎麼做音檔時頻域轉換，結果程式瘋狂報錯。資料安全性的問題，似乎是比較之間的問題，不是說一般使用者資料不重要，雖然一般人的個資已經是很容易就癑取得的東西，而是看和誰比，大公司在開發的工具，可能是極巨商業利益，禁止隨意上傳資料到別人的產品，在資本社會中應該是蠻合理的，畢竟也難保人家用其他方法獲取使用者上傳的資料。

**Reply by pudding0803** [reply]
偏大型或罕見的問題似乎都滿容易會讓他給予錯誤資訊的😅。不過我後來有發現，你只要問他是不是正確的，他就會自己判斷他剛剛說的對不對，並且進行更正，雖然有時候要更正好幾次就是了🤣

是的🥹，畢竟據說我們的資料早就都外洩過了QQ，但公司更會牽扯到金錢與利益的問題，所以禁止也是滿合理的

參考文章：[到底是誰洩漏台灣2300萬筆個資？數發部為何攔阻不了？](https://www.cw.com.tw/article/5123412)

**Comment by aa103967622** [comment]
Chatgpt本身就詢問你的提問紀錄是否要回傳並提供給chatgpt訓練使用，否則這些紀錄會在好像是一個月左右被刪除，應該大部分人不想丟失紀錄，都會打勾同意吧？chatgpt是個強大的工具，而不是你的完美代打！只要你針對你要提問的問題給予限制，比如說我要求它不能簡單條列式回答，而是針對原理說明並做舉例之類的，像這樣在問法上有點小巧思，其實你快速能得到很多精準的資料，儘量不要把整個問題丟給chatgpt處理，不僅容易把更多資料洩漏出去，你也不能從解決問題的過程中學到東西，是得不償失的。

**Reply by pudding0803** [reply]
原來紀錄會被刪出，之前沒有特別注意😮

是的！如何問出精準的問題，讓他可以理解並回覆你想要的答案也是一種技巧，確實只要知道怎麼針對特定問題去進行詢問，而不是整個丟上去，大概就可以非常有效地預防資料外洩了！
